### ISSUE TITLE
BigQuery's "SELECT AS STRUCT" mangled by fix

### ISSUE THREAD

**Issue by daaain** [sqlfluff/sqlfluff#1082]
With this query as a simple case (saved as `models/test.sql`):

```sql
select array(
  select as struct
    b as bb,
    a as aa
  from example
)
```

The issue is present even without the `select array()` bit, but this above is a real use case.

## Expected Behaviour

Pass without errors and not get mangled as it's valid BigQuery SQL, see: https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#select_as_struct

## Observed Behaviour

I get:

```
== [models/test.sql] FAIL
L:   1 | P:   1 | L036 | Select targets should be on a new line unless there is
                       | only one select target.
L:   1 | P:   8 | L034 | Use wildcards then simple select targets before
                       | calculations and aggregates.
```

And the "fixed" SQL is:

```sql
select array(
  select
     a as aa,
    as struct
    b as bb
  from example
)
```

## Steps to Reproduce

```sh
sqlfluff fix --force models/test.sql
```

## Version

```sh
app@27fc52961e2b:/app$ sqlfluff --version
sqlfluff, version 0.5.6
app@27fc52961e2b:/app$ python --version
Python 3.7.10
```

## Configuration

```
[sqlfluff]
templater = dbt
dialect = bigquery
exclude_rules = L031

[sqlfluff:rules]
tab_space_size = 2
max_line_length = 100

[sqlfluff:rules:L030]
capitalisation_policy = lower

```

**Comment by barrywhart** [comment]
This _may_ be two issues in one:
* Dialect support for parsing `select as struct`
* Ensuring that lint rule(s) don't get confused by `as` being used outside of column and table aliases

Regarding the parse issue, apparently BigQuery has two different syntaxes for doing the same thing.

SQLFluff supports this:
```
select struct(
  '1' as bb,
  2 as aa)
```

But not this:
```
select as struct
  '1' as bb,
  2 as aa
```

**Comment by daaain** [comment]
Thanks a lot for the reply and the fix!

Didn't know about the `select struct()` syntax, but looking at it I actually prefer it as it's consistent with `select array()` and not ambiguous with the more common usage of `as`, so might even switch to it...

**Comment by barrywhart** [comment]
Working on this project has definitely exposed me to many areas of BigQuery I'd never heard of otherwise!
